linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -984,35 +1047,30 @@ final class Analyzer(config: CommonPhaseConfig, initial: Boolean,
         val info = new TopLevelExportInfo(className, tle)
         info.reach()
 
-        _topLevelExportInfos.get(key).fold[Unit] {
-          _topLevelExportInfos.put(key, info)
-        } { other =>
-          _errors += ConflictingTopLevelExport(tle.moduleID, tle.exportName, List(info, other))
+        _topLevelExportInfos.put(key, info).foreach { other =>
+          addError(ConflictingTopLevelExport(tle.moduleID, tle.exportName, List(info, other)))
         }
       }
     }
 
     def accessModule()(implicit from: From): Unit = {
       if (!isAnyModuleClass) {
-        _errors += NotAModule(this, from)
-      } else if (!isModuleAccessed) {
-        isModuleAccessed = true
-
-        instantiated()
+        addError(NotAModule(this, from))
+      } else if (!isModuleAccessed.getAndSet(true)) {
+        instantiated() // TODO: Shouldn't we always add the from?
         if (isScalaClass)
           callMethodStatically(MemberNamespace.Constructor, NoArgConstructorName)
       }
     }
 
     def instantiated()(implicit from: From): Unit = {
-      instantiatedFrom ::= from
+      _instantiatedFrom.updateAndGet(from :: _)
 
-      /* TODO? When the second line is false, shouldn't this be a linking error
+      /* TODO? When the first line is false, shouldn't this be a linking error
        * instead?
        */
-      if (!isInstantiated &&
-          (isScalaClass || isJSClass || isNativeJSClass)) {
-        isInstantiated = true
+      if ((isScalaClass || isJSClass || isNativeJSClass) &&
+          !_isInstantiated.getAndSet(true)) {
```

Comment:
Scalastyle fails on this line, claiming "Boolean expression can be simplified.

Relevant code:
https://github.com/scalastyle/scalastyle/blob/v1.0.0/src/main/scala/org/scalastyle/scalariform/SimplifyBooleanExpressionChecker.scala

I do not understand what is happening. From how I read the relevant code, it should trigger on the following operators being used with literal boolean values: `! != == && ||`.

Reply:
Note: Switching to infix notation for `getAndSet` or method notation for `&&` makes the error go away. But maybe we should just deactivate the check?

Reply:
I'd rather deactivate the check than arbitrarily switching notation.